app/services/importers/create_new_ecf_cohort.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Importers
+  class CreateNewECFCohort
+    def call
+      logger.info "CreateNewECFCohort: Started!"
+      check_csvs!
+
+      run_ecf_cohort_scripts
+      logger.info "CreateNewECFCohort: Finished!"
+    end
+
+  private
+
+    attr_reader :cohort_csv, :cohort_lead_provider_csv, :contract_csv, :schedule_csv, :statement_csv, :logger
+
+    def initialize(cohort_csv:, cohort_lead_provider_csv:, contract_csv:, schedule_csv:, statement_csv:, logger: Rails.logger)
+      @cohort_csv = cohort_csv
+      @cohort_lead_provider_csv = cohort_lead_provider_csv
+      @contract_csv = contract_csv
+      @schedule_csv = schedule_csv
+      @statement_csv = statement_csv
+      @logger = logger
+    end
+
+    def check_csvs!
+      return if cohort_csv.present? && cohort_lead_provider_csv.present? && contract_csv.present? && schedule_csv.present? && statement_csv.present?
+
+      raise "All scripts need to be present to create a new ECF cohort"
+    end
+
+    def run_ecf_cohort_scripts
```

Comment:
@cwrw Do you think we need some sort of rollback if any of the steps fails mid-process for some reason?

Reply:
@leandroalemao good question, each script we call is wrapped within a transaction block, so anything that isn't valid will rollback. We will also test the data before running in prod so there are no surprises